boa_parser/src/parser/expression/primary/object_initializer/mod.rs

```diff
@@ -292,8 +292,14 @@ where
 
         //  PropertyName[?Yield, ?Await] : AssignmentExpression[+In, ?Yield, ?Await]
         if cursor.next_if(Punctuator::Colon, interner)?.is_some() {
-            let value = AssignmentExpression::new(None, true, self.allow_yield, self.allow_await)
+            let name = property_name
+                .literal()
+                .filter(|name| name != &Sym::__PROTO__)
```

Comment:
Since `Sym` is `Copy`, I would suggest to either dereference `name`, or use `copied()`:
```suggestion
                .filter(|name| *name != Sym::__PROTO__)
```

Even though I guess it's equivalent, it more clearly shows intent of comparing the symbol value, and not the reference.